e2e/policy/policy_test.go

```diff
@@ -0,0 +1,177 @@
+
+	t.Run("pod cannot join after it was removed from the manifest", func(t *testing.T) {
+		require := require.New(t)
+
+		ctx, cancel := context.WithTimeout(context.Background(), 1*time.Minute)
+		defer cancel()
+
+		c := kubeclient.NewForTest(t)
+
+		t.Log("Waiting for deployments")
+		require.NoError(c.WaitFor(ctx, kubeclient.Deployment{}, ct.Namespace, opensslBackend))
+		require.NoError(c.WaitFor(ctx, kubeclient.Deployment{}, ct.Namespace, opensslFrontend))
+
+		// get the attestation failures before removing a policy
+		initialFailures := getFailures(ctx, t, ct)
+
+		t.Log("Initial failures:", initialFailures)
+
+		// parse the manifest
+		manifestBytes, err := os.ReadFile(path.Join(ct.WorkDir, "manifest.json"))
+		require.NoError(err)
+		var m manifest.Manifest
+		require.NoError(json.Unmarshal(manifestBytes, &m))
+
+		// Remove a policy from the manifest.
+		for policyHash := range m.Policies {
+			if slices.Contains(m.Policies[policyHash], opensslFrontend) {
+				delete(m.Policies, policyHash)
+			}
+		}
+
+		// write the new manifest
+		manifestBytes, err = json.Marshal(m)
+		require.NoError(err)
+		require.NoError(os.WriteFile(path.Join(ct.WorkDir, "manifest.json"), manifestBytes, 0o644))
+
+		// parse the original resources
+		resourceBytes, err := os.ReadFile(path.Join(ct.WorkDir, "resources.yaml"))
+		require.NoError(err)
+		r, err := kubeapi.UnmarshalUnstructuredK8SResource(resourceBytes)
+		require.NoError(err)
+
+		// remove everything from the openssl-frontend
+		newResources := make([]*unstructured.Unstructured, 0, len(r))
+		for _, resource := range r {
+			name := resource.GetName()
+			require.NoError(err)
+			if strings.Contains(name, opensslFrontend) {
+				continue
+			}
+			newResources = append(newResources, resource)
+		}
+
+		// write the new resources yaml
+		resourceBytes, err = kuberesource.EncodeUnstructured(newResources)
+		require.NoError(err)
+		require.NoError(os.WriteFile(path.Join(ct.WorkDir, "resources.yaml"), resourceBytes, 0o644))
+
+		// set the new manifest
+		ct.Set(t)
+
+		// restart the deployments
+		require.NoError(c.Restart(ctx, kubeclient.Deployment{}, ct.Namespace, opensslFrontend)) // not waiting since it would fail
+		require.NoError(c.Restart(ctx, kubeclient.Deployment{}, ct.Namespace, opensslBackend))
+		require.NoError(c.WaitFor(ctx, kubeclient.Deployment{}, ct.Namespace, opensslBackend))
+
+		// wait for the init container of the openssl-frontend pod to enter the running state
+		ready := false
+		for !ready {
+			time.Sleep(1 * time.Second)
+			pods, err := ct.Kubeclient.PodsFromDeployment(ctx, ct.Namespace, opensslFrontend)
+			require.NoError(err)
+			require.NotEmpty(pods, "pod not found: %s/%s", ct.Namespace, opensslFrontend)
+			require.NotEmpty(pods[0].Status.InitContainerStatuses, "pod doesn't contain init container statuses: %s/%s", ct.Namespace, opensslFrontend)
+			ready = pods[0].Status.InitContainerStatuses[0].State.Running != nil
+		}
```

Comment:
If we restructure `kubeclient.WaitFor` a bit, we might be able to use that instead of handrolling the waiting loop here. However, let's leave that as a follow-up for now.